### PR TITLE
[ios] Potential fix for over retain/release issues with adding layers…

### PR DESCRIPTION
**Pull request by julianrex**

### CODE REVIEW

File: platform/ios/ios.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1908,7 +1920,6 @@
 				DA8848231CBAFA6200AB86E3 /* MGLOfflineStorage_Private.h in Headers */,
 				404326891D5B9B27007111BD /* MGLAnnotationContainerView_Private.h in Headers */,
 				CA55CD41202C16AA00CE7095 /* MGLCameraChangeReason.h in Headers */,
-				1FB7DAAF1F2A4DBD00410606 /* MGLVectorSource+MGLAdditions.h in Headers */,
```
Comment (julianrex) [inline platform/ios/ios.xcodeproj/project.pbxproj]: Not sure what happened here...

**Comment by julianrex** [comment]
testAddingLayerOfTypeABeforeRemovingLayerOfTypeBWithSameIdentifier is failing.

**Comment by akitchen** [comment]
These changes look good; I'd like to spend some time with the code checked out locally but let's consolidate into the one branch as this PR aims to do (we can rename that other PR or close it and open a new one).

**Comment by akitchen** [comment]
Up to you @julianrex if you'd like to keep this branch around.

**Pull request merged**